code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +626,237 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+	}
+
+	// now being to configure each object in the group (mines multiple)
+
+	for (int a = 0; a < number_of_objects; a++) { // make an a-b 2d grid from the mines
+		offsetAxisA += (mine_dist * a); // add a new row to the grid
+		for (int b = 0; b < number_of_objects; b++) { // for each increment along the a plane, add mines along b plane
+			offsetAxisB += (mine_dist * b);  // for each new row populate the column
+			
+			/** Now convert the grid (a,b) to the relavenat formation ie. (x,y) or (z,y) etc **/
+			auto ojxyz = xwi_determine_mine_formation_position(oj, objectPosX, objectPosY, objectPosZ, offsetAxisA, offsetAxisB);
+			vm_vec_scale(&ojxyz, 1000); // units are in kilometers (after processing by xwinglib which handles the
+										// factor of 160), so scale them up
+
+			pobj.orient = orient;
+			pobj.pos = ojxyz;
+
+			// regular name, regular suffix
+
+			char base_name[NAME_LENGTH];
+			char suffix[NAME_LENGTH];
+			char *newArray = new char[std::strlen(base_name) + std::strlen(suffix) + 1];
+			strcpy_s(base_name, object_type);
+
+			end_string_at_first_hash_symbol(base_name);
+
+			// find lowest unique n for the suffix
+			int n = 1;
+			sprintf(suffix, NOX(" %d"), n);
+
+			// objects don't have designations so they are limited to the name from the object_type and suffix
+			// names will be hidden in game so to avoid iterating multiple times, allow for suffix 999 (+4)
+			int char_overflow = static_cast<int>(strlen(base_name) + 4) - (NAME_LENGTH - 1);
+			if (char_overflow > 0) {
+				base_name[strlen(base_name) - static_cast<size_t>(char_overflow)] = '\0';
+			}
+			char croppedName[NAME_LENGTH];
+			strcpy_s(croppedName, base_name); // save base_name incase suffix needs changed
+
+			strcat_s(croppedName, suffix);
+			// Now check the name against the objectSet of names
+			auto fin = objectSet.end();
+			while (objectSet.find(croppedName) != fin) {
+				n++;
+				sprintf(suffix, NOX(" %d"), n);
+				strcpy_s(croppedName, base_name);
+				strcat_s(croppedName, suffix);
+			}
+
+			strcpy_s(pobj.name, croppedName);
+			
+			objectSet.insert(pobj.name);  // add the new name to the objectSet
+			SCP_totitle(pobj.name);
+			Parse_objects.push_back(pobj);
+
+			/** Is any of this needed below?
+
+			pobj.arrival_cue = arrival_cue;
+			pobj.arrival_delay = fg->arrivalDelay;
+			pobj.arrival_location = fg->arriveByHyperspace ? ARRIVE_AT_LOCATION : ARRIVE_FROM_DOCK_BAY;
+			pobj.arrival_anchor = xwi_determine_anchor(xwim, fg);
+			pobj.departure_location = fg->departByHyperspace ? DEPART_AT_LOCATION : DEPART_AT_DOCK_BAY;
+			pobj.departure_anchor = pobj.arrival_anchor;
+
+			// if a ship doesn't have an anchor, make sure it is at-location
+			// (flight groups present at mission start will have arriveByHyperspace set to false)
+			if (pobj.arrival_anchor < 0)
+				pobj.arrival_location = ARRIVE_AT_LOCATION;
+			if (pobj.departure_anchor < 0)
+				pobj.departure_location = DEPART_AT_LOCATION;
```

Comment:
Some of this is needed, but it should use default values.  The arrival cue should be `Locked_sexp_true`, the departure cue should be `Locked_sexp_false`, and the locations should be at-location.  The delays and anchors aren't needed.